File: VB/PivotGridCustomization.Module.Win/Updater.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Data.Filtering
Imports DevExpress.ExpressApp.PivotChart
Imports DevExpress.ExpressApp.PivotChart.Win

Namespace PivotGridCustomization.Module.Win
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal session As Session, ByVal currentDBVersion As Version)
			MyBase.New(session, currentDBVersion)
		End Sub
		Private Sub CreateTestAnalysis()
			Dim testAnalysis As Analysis = Session.FindObject(Of Analysis)(CriteriaOperator.Parse("Name='Test Analysis'"))
			If testAnalysis Is Nothing Then
				testAnalysis = New Analysis(Session)
				testAnalysis.Name = "Test Analysis"
				testAnalysis.ObjectTypeName = GetType(TestObject).FullName
				Using control As New AnalysisControlWin()
					control.DataSource = New AnalysisDataSource(testAnalysis, New XPCollection(Of TestObject)(testAnalysis.Session))
					control.Fields("Name").Area = DevExpress.XtraPivotGrid.PivotArea.RowArea
					control.Fields("Value").Area = DevExpress.XtraPivotGrid.PivotArea.DataArea
					control.Fields("AnotherValue").Area = DevExpress.XtraPivotGrid.PivotArea.ColumnArea
					PivotGridSettingsHelper.SavePivotGridSettings(New PivotGridControlSettingsStore((control.PivotGrid)), testAnalysis)
				End Using
				testAnalysis.Save()
			End If
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()
			CreateTestAnalysis()
		End Sub
	End Class
End Namespace

File: CS/PivotGridCustomization.Module.Win/Updater.cs
using System;

using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Data.Filtering;
using DevExpress.ExpressApp.PivotChart;
using DevExpress.ExpressApp.PivotChart.Win;

namespace PivotGridCustomization.Module.Win {
    public class Updater : ModuleUpdater {
        public Updater(Session session, Version currentDBVersion) : base(session, currentDBVersion) { }
        private void CreateTestAnalysis() {
            Analysis testAnalysis = Session.FindObject<Analysis>(CriteriaOperator.Parse("Name='Test Analysis'"));
            if (testAnalysis == null) {
                testAnalysis = new Analysis(Session);
                testAnalysis.Name = "Test Analysis";
                testAnalysis.ObjectTypeName = typeof(TestObject).FullName;
                using (AnalysisControlWin control = new AnalysisControlWin()) {
                    control.DataSource = new AnalysisDataSource(testAnalysis, new XPCollection<TestObject>(testAnalysis.Session));
                    control.Fields["Name"].Area = DevExpress.XtraPivotGrid.PivotArea.RowArea;
                    control.Fields["Value"].Area = DevExpress.XtraPivotGrid.PivotArea.DataArea;
                    control.Fields["AnotherValue"].Area = DevExpress.XtraPivotGrid.PivotArea.ColumnArea;
                    PivotGridSettingsHelper.SavePivotGridSettings(new PivotGridControlSettingsStore((control.PivotGrid)), testAnalysis);
                }
                testAnalysis.Save();
            }
        }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            CreateTestAnalysis();
        }
    }
}
